src/test/scala/it/unibo/authsim/library/dsl/attack/builders/BruteForceAttackBuilderTest.scala

```diff
@@ -14,53 +17,60 @@ class BruteForceAttackBuilderTest extends AnyWordSpec:
 
     def getStatistics: Statistics = this.statistics
 
+
+    //TODO Alex appena abbiamo le policy degli algoritmi togli sta roba, te l'ho messa se no non compilava più nulla
+    @Mock
+    abstract class AlgorithmPolicy extends Policy
+    @Mock
+    val algorithmPolicy = mock[AlgorithmPolicy]
+
```

Comment:
The only thing I don't understand is why you put this mocks inside the private class of the consumer.

Reply:
Because in your test class you use the `UserProvider` which contains `CryptoInformation` that requires an `algorithmPolicy` (if I am not wrong we decide so in the last meeting), however, is policy does not exist yet.
For this reason to make the test run despite this issue I created a fake class AlgorithmPolicy.

Reply:
Ok, now I see the problem: the indentation is wrong: the private class should finish after the `consume` override (and after that you can put the mocks).

The correct indentation should look like this (with the `TestStatisticsConsumer` and the mock declarations at the same level of indentation`):
```
  private class TestStatisticsConsumer extends StatisticsConsumer:
    private var statistics: Statistics = Statistics.zero

    def getStatistics: Statistics = this.statistics

    override def consume(consumable: Statistics): Unit =
      this.statistics = consumable
      println("Attempts: " + consumable.attempts)
      println("Elapsed time: " + consumable.elapsedTime.toMillis + " ms")
      println("Breached credentials: " + consumable.successfulBreaches.map(u => u.username + " - " + u.password).reduceOption((u1, u2) => u1 + "\n" + u2).getOrElse("No credentials breached"))

  //TODO Alex appena abbiamo le policy degli algoritmi togli sta roba, te l'ho messa se no non compilava più nulla
  @Mock
  abstract class AlgorithmPolicy extends Policy
  @Mock
    val algorithmPolicy = mock[AlgorithmPolicy]
```

I suspect the same situation is in the `DictionaryAttackBuilderTest`.